oneview/resource_server_profile.go

```diff
@@ -95,49 +88,36 @@ func resourceServerProfile() *schema.Resource {
 func resourceServerProfileCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	serverProfileTemplate, err := config.ovClient.GetProfileTemplateByName(d.Get("template").(string))
-	if err != nil || serverProfileTemplate.URI.IsNil() {
-		return fmt.Errorf("Could not find Server Profile Template\n%+v", d.Get("template").(string))
+	serverProfile := ov.ServerProfile{
+		Type: d.Get("type").(string),
+		Name: d.Get("name").(string),
 	}
-	var serverHardware ov.ServerHardware
+
 	if val, ok := d.GetOk("hardware_name"); ok {
-		serverHardware, err = config.ovClient.GetServerHardwareByName(val.(string))
+		serverHardware, err := config.ovClient.GetServerHardwareByName(val.(string))
 		if err != nil {
 			return err
 		}
-	} else {
-		var hwFilters = []string{}
-		for _, filter := range d.Get("hw_filter").([]interface{}) {
-			hwFilters = append(hwFilters, filter.(string))
-		}
-		serverHardware, err = getServerHardware(config, serverProfileTemplate, hwFilters)
-		if err != nil {
-			return err
+		if serverHardware.PowerState != "off" {
+			return errors.New("Server Hardware must be powered off to assign to the server profile")
 		}
+		serverProfile.ServerHardwareURI = serverHardware.URI
 	}
 
-	profileType := d.Get("type")
-	if profileType == "ServerProfileV6" {
```

Comment:
Are types pre `ServerProfileV9` not supported anymore? It'd be worth calling out in the docs what types are supported.

Reply:
yes all the types are supported. But i'm not sure I understand the logic of the code in the master branch.

`CreateProfileFromTemplateWithI3S` is called if the type is `V6` and `CreateProfileFromTemplate` is called otherwise. This shouldn't be so, as this is not nessarily the case always. I must be able to call `CreateProfileFromTemplate` even when the type is `V6`. I have added the code which supports this.